### PR TITLE
Fix Marian Unicode casing across byte-fallback tokens

### PR DESCRIPTION
**Pull request by mustjab**
## Summary

Fix Marian Unicode case restoration when a UTF-8 code point is split across byte-fallback tokens.

The case encoder can emit a titlecase marker followed by multiple byte pieces. `SpmUgmDecoder::Id2Token` previously processed each piece independently and emitted incomplete bytes unchanged. Concatenating those bytes reconstructed the lowercase character, but it was too late to apply the active casing mode.

For example, `T` followed by `C3 A9` must decode to U+00C9 (Latin capital E with acute), not U+00E9. Similarly, `T` followed by `D0 B1` must produce U+0411 rather than U+0431. ASCII titlecasing already worked.

## Changes

- Reuse the per-stream `incomplete_utf8_` buffer until a complete code point is available, then apply the existing Marian casing rules.
- Preserve pending bytes at special tokens, unknown IDs, and the end of full-sequence decoding.
- Validate multibyte candidates with `ustring::ValidateUTF8` before casing, preserving invalid encodings.
- Use fresh state and advance the input pointer for each row of full batch decoding.
- Avoid copying the vocabulary piece when no bytes are pending.
- Guard trailing-space trimming when full decoding produces an empty string.
- Add model-free byte-vocabulary tests covering two-, three-, and four-byte letters, unmarked text, casing modes, word boundaries, malformed/truncated bytes, full decoding without EOS, and interleaved independent streams.

The non-Marian unigram decoding path remains unchanged for normal input. This does not change encoding, vocabulary IDs, model weights, or search settings.

## Validation

Built `extensions_shared` and `pp_api_test` on Windows x64, Release (Visual Studio 2026).

All 18 selected tests pass:

```text
MarianByteFallbackTest.*
MarianId2TokenTest.*
OrtxTokenizerTest.MarianTokenizer
OrtxTokenizerTest.MarianTokenizer2
OrtxTokenizerTest.SpmUgmTokenizer
OrtxTokenizerTest.ClipTokenizer
CApiTest.StreamApiTest
```

Also verified the original model-free reproduction through both `OrtxDetokenize1D` and `OrtxDetokenizeCached`: Unicode titlecase cases fail before the fix and pass afterward; ASCII and lowercase controls pass in both versions.

Both Copilot findings were fixed and regression-tested, and the supplemental invalid-ID test was added. Copilot's re-review of the runtime fixes recommended approval with no new comments. Linux validation has not been run locally; cross-platform CI remains required.

### CODE REVIEW

File: operators/tokenizer/ugm_kernels.hpp [hunk 1]
```diff
@@ -788,13 +790,16 @@ class SpmUgmDecoder {
         text += token;
       }
 
+      if (state != nullptr) {
+        text += std::exchange(state->incomplete_utf8_, {});
+      }
```
[the remark this answers is not in the document]
Comment (mustjab) [reply operators/tokenizer/ugm_kernels.hpp:796]: Fixed in 3f5ccbf: each full-decoding row now owns a fresh state, and the input pointer advances after each row. FullDecodeResetsStateBetweenRows covers distinct rows ending in a titlecase marker, incomplete UTF-8, and an uppercase run, followed by lowercase rows. All 17 selected tokenizer tests pass on Windows x64 Release.

File: operators/tokenizer/ugm_kernels.hpp [hunk 2]
```diff
@@ -831,6 +836,11 @@ class SpmUgmDecoder {
     } else {
       return false;
     }
+    for (size_t index = 1; index < char_len; ++index) {
+      if ((static_cast<unsigned char>(utf8[index]) & 0xC0) != 0x80) {
+        return false;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (mustjab) [reply operators/tokenizer/ugm_kernels.hpp]: Fixed in 3f5ccbf: multibyte candidates must pass ustring::ValidateUTF8 before casing is applied. InvalidCodepointsAreNotCaseConverted covers two-, three-, and four-byte overlong encodings, a surrogate, an out-of-range code point, and a noncharacter through full and incremental decoding. All 17 selected tokenizer tests pass on Windows x64 Release.

**Review by kunal-vaishnavi — APPROVED** [review]
